### ISSUE TITLE
ansible-galaxy install on core 2.14.0 stuck indefinitely

### ISSUE THREAD

**Issue by guidograzioli** [linux-system-roles/linux-system-roles.github.io#102]
The command line

    ansible-galaxy collection install fedora.linux_system_roles --force-with-deps -vvv

on ansible-core>=2.15 and ansible-core==2.14.14 (latest at the time of writing) works correcly,

Trying on a ansible-core 2.14.0 virtualenv, it gets stuck indefinitely (I interrupted after ~10 minutes)

```
(core-2.14) guido@fedora:~/Development/redhat/github-actions$ ansible-galaxy collection install fedora.linux_system_roles --force-with-deps -vvv
ansible-galaxy [core 2.14.0]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/guido/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/guido/Development/virtualenv/core-2.14/lib/python3.11/site-packages/ansible
  ansible collection location = /home/guido/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/guido/Development/virtualenv/core-2.14/bin/ansible-galaxy
  python version = 3.11.8 (main, Feb 28 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)] (/home/guido/Development/virtualenv/core-2.14/bin/python)
  jinja version = 3.1.3
  libyaml = True
Using /etc/ansible/ansible.cfg as config file
Starting galaxy collection install process
Found installed collection middleware_automation.redhat_csp_download:1.2.2 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/redhat_csp_download'
Found installed collection middleware_automation.janus:1.0.0 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/janus'
Found installed collection middleware_automation.jws:1.2.5 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/jws'
Found installed collection middleware_automation.jbcs:1.0.4 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/jbcs'
Found installed collection middleware_automation.infinispan:1.3.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/infinispan'
Found installed collection middleware_automation.keycloak:2.1.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/keycloak'
Found installed collection middleware_automation.wildfly:1.5.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/wildfly'
Found installed collection middleware_automation.amq:2.0.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/amq'
Found installed collection middleware_automation.common:1.2.0 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/common'
Found installed collection ansible.netcommon:5.1.1 at '/home/guido/.ansible/collections/ansible_collections/ansible/netcommon'
Found installed collection ansible.utils:2.10.3 at '/home/guido/.ansible/collections/ansible_collections/ansible/utils'
Found installed collection ansible.posix:1.5.4 at '/home/guido/.ansible/collections/ansible_collections/ansible/posix'
Found installed collection community.docker:3.4.8 at '/home/guido/.ansible/collections/ansible_collections/community/docker'
Found installed collection community.fqcn_migration:1.0.2 at '/home/guido/.ansible/collections/ansible_collections/community/fqcn_migration'
Found installed collection community.aws:7.1.0 at '/home/guido/.ansible/collections/ansible_collections/community/aws'
Found installed collection community.general:8.4.0 at '/home/guido/.ansible/collections/ansible_collections/community/general'
Found installed collection guidograzioli.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/kubernetes.kubevirt'
Found installed collection redhat.eap:1.3.4 at '/home/guido/.ansible/collections/ansible_collections/redhat/eap'
Found installed collection redhat.jbcs:1.0.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/jbcs'
Found installed collection redhat.sso:2.0.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/sso'
Found installed collection redhat.runtimes_common:1.1.3 at '/home/guido/.ansible/collections/ansible_collections/redhat/runtimes_common'
Found installed collection redhat.data_grid:1.3.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/data_grid'
Found installed collection redhat.rhbk:2.1.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/rhbk'
Found installed collection redhat.jws:2.1.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/jws'
Found installed collection redhat.amq_broker:2.0.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/amq_broker'
Found installed collection amazon.aws:7.3.0 at '/home/guido/.ansible/collections/ansible_collections/amazon/aws'
Found installed collection azure.azcollection:1.16.0 at '/home/guido/.ansible/collections/ansible_collections/azure/azcollection'
Found installed collection containers.podman:1.12.0 at '/home/guido/.ansible/collections/ansible_collections/containers/podman'
Found installed collection kubernetes.core:2.4.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/core'
Found installed collection guidograzioli.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old2'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old3'
Found installed collection kubevirt.core:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/guidograzioli/kubevirt'
Found installed collection kubevirt.core:1.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubevirt/core'
Found installed collection skupper.network:1.2.1 at '/home/guido/.ansible/collections/ansible_collections/skupper/network'
Found installed collection infra.rhbk_datagrid_aws:1.0.0 at '/home/guido/.ansible/collections/ansible_collections/infra/rhbk_datagrid_aws'
Found installed collection fedora.linux_system_roles:1.75.3 at '/home/guido/.ansible/collections/ansible_collections/fedora/linux_system_roles'
Process install dependency map
Opened /home/guido/.ansible/galaxy_token
^C [ERROR]: User interrupted execution
```

I noticed the behaviour in our github actions CI, where we were installing ansible-core in a matrix with `==`, and using `~=2.14` fixed it

**Comment by richm** [comment]
Can you provide the exact steps you used to create your venv and install ansible in the venv?

**Comment by guidograzioli** [comment]
Sure, on fedora 39: 

```
guido@fedora:~/Development/redhat/common$ virtualenv -p python3.11 /home/guido/Development/virtualenv/2.14
created virtual environment CPython3.11.8.final.0-64 in 119ms
  creator CPython3Posix(dest=/home/guido/Development/virtualenv/2.14, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(extra_search_dir=/usr/share/python-wheels,download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/guido/.local/share/virtualenv)
    added seed packages: pip==23.2.1, setuptools==67.7.2, wheel==0.40.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
```
```
guido@fedora:~/Development/redhat/common$ source /home/guido/Development/virtualenv/2.14/bin/activate
```

```
(2.14) guido@fedora:~/Development/redhat/common$ pip install 'ansible-core==2.14.0'
Collecting ansible-core==2.14.0
  Obtaining dependency information for ansible-core==2.14.0 from https://files.pythonhosted.org/packages/ef/ef/0448f37c05769497e919c04eb668f379fbaeaf0c3188c01cd09c92253f76/ansible_core-2.14.0-py3-none-any.whl.metadata
  Downloading ansible_core-2.14.0-py3-none-any.whl.metadata (7.4 kB)
Collecting jinja2>=3.0.0 (from ansible-core==2.14.0)
[... cut ...]
Using cached packaging-24.0-py3-none-any.whl (53 kB)
Using cached cffi-1.16.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (464 kB)
Using cached MarkupSafe-2.1.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (28 kB)
Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Installing collected packages: resolvelib, PyYAML, pycparser, packaging, MarkupSafe, jinja2, cffi, cryptography, ansible-core
rsionSuccessfully installed MarkupSafe-2.1.5 PyYAML-6.0.1 ansible-core-2.14.0 cffi-1.16.0 cryptography-42.0.5 jinja2-3.1.3 packaging-24.0 pycparser-2.21 resolvelib-0.8.1

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: pip install --upgrade pip
```

```
(2.14) guido@fedora:~/Development/redhat/common$ ansible --version
ansible [core 2.14.0]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/guido/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages/ansible
  ansible collection location = /home/guido/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/guido/Development/virtualenv/2.14/bin/ansible
  python version = 3.11.8 (main, Feb 28 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)] (/home/guido/Development/virtualenv/2.14/bin/python)
  jinja version = 3.1.3
  libyaml = True
```

```
(2.14) guido@fedora:~/Development/redhat/common$ ansible-galaxy collection install --no-cache fedora.linux_system_roles --force -vvv
ansible-galaxy [core 2.14.0]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/guido/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages/ansible
  ansible collection location = /home/guido/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/guido/Development/virtualenv/2.14/bin/ansible-galaxy
  python version = 3.11.8 (main, Feb 28 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)] (/home/guido/Development/virtualenv/2.14/bin/python)
  jinja version = 3.1.3
  libyaml = True
Using /etc/ansible/ansible.cfg as config file
Starting galaxy collection install process
Found installed collection middleware_automation.redhat_csp_download:1.2.2 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/redhat_csp_download'
Found installed collection middleware_automation.janus:1.0.0 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/janus'
Found installed collection middleware_automation.jws:1.2.5 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/jws'
Found installed collection middleware_automation.jbcs:1.0.4 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/jbcs'
Found installed collection middleware_automation.infinispan:1.3.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/infinispan'
Found installed collection middleware_automation.keycloak:2.1.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/keycloak'
Found installed collection middleware_automation.wildfly:1.5.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/wildfly'
Found installed collection middleware_automation.amq:2.0.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/amq'
Found installed collection middleware_automation.common:1.2.0 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/common'
Found installed collection ansible.netcommon:5.1.1 at '/home/guido/.ansible/collections/ansible_collections/ansible/netcommon'
Found installed collection ansible.utils:2.10.3 at '/home/guido/.ansible/collections/ansible_collections/ansible/utils'
Found installed collection ansible.posix:1.5.4 at '/home/guido/.ansible/collections/ansible_collections/ansible/posix'
Found installed collection community.docker:3.4.8 at '/home/guido/.ansible/collections/ansible_collections/community/docker'
Found installed collection community.fqcn_migration:1.0.2 at '/home/guido/.ansible/collections/ansible_collections/community/fqcn_migration'
Found installed collection community.aws:7.1.0 at '/home/guido/.ansible/collections/ansible_collections/community/aws'
Found installed collection community.general:8.4.0 at '/home/guido/.ansible/collections/ansible_collections/community/general'
Found installed collection guidograzioli.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/kubernetes.kubevirt'
Found installed collection redhat.eap:1.3.4 at '/home/guido/.ansible/collections/ansible_collections/redhat/eap'
Found installed collection redhat.jbcs:1.0.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/jbcs'
Found installed collection redhat.sso:2.0.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/sso'
Found installed collection redhat.data_grid:1.3.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/data_grid'
Found installed collection redhat.rhbk:2.1.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/rhbk'
Found installed collection redhat.jws:2.1.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/jws'
Found installed collection redhat.amq_broker:2.0.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/amq_broker'
Found installed collection redhat.runtimes_common:1.2.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/runtimes_common'
Found installed collection redhat.rhel_system_roles:1.23.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/rhel_system_roles'
Found installed collection amazon.aws:7.3.0 at '/home/guido/.ansible/collections/ansible_collections/amazon/aws'
Found installed collection azure.azcollection:1.16.0 at '/home/guido/.ansible/collections/ansible_collections/azure/azcollection'
Found installed collection containers.podman:1.12.0 at '/home/guido/.ansible/collections/ansible_collections/containers/podman'
Found installed collection kubernetes.core:2.4.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/core'
Found installed collection guidograzioli.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old2'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old3'
Found installed collection kubevirt.core:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/guidograzioli/kubevirt'
Found installed collection kubevirt.core:1.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubevirt/core'
Found installed collection skupper.network:1.2.1 at '/home/guido/.ansible/collections/ansible_collections/skupper/network'
Found installed collection infra.rhbk_datagrid_aws:1.0.0 at '/home/guido/.ansible/collections/ansible_collections/infra/rhbk_datagrid_aws'
Found installed collection fedora.linux_system_roles:1.75.3 at '/home/guido/.ansible/collections/ansible_collections/fedora/linux_system_roles'
Process install dependency map
Opened /home/guido/.ansible/galaxy_token
[.. stuck here..]
```

**Comment by guidograzioli** [comment]
Then in same venv:

```
(2.14) guido@fedora:~/Development/redhat/common$ pip install -U 'ansible-core~=2.14.0'
Collecting ansible-core~=2.14.0
  Obtaining dependency information for ansible-core~=2.14.0 from https://files.pythonhosted.org/packages/e3/32/43baf6188500e1fb00e6ba10f5b1c9d4d8879802823a11b22874d95b8a2e/ansible_core-2.14.14-py3-none-any.whl.metadata
  Downloading ansible_core-2.14.14-py3-none-any.whl.metadata (6.9 kB)
Requirement already satisfied: jinja2>=3.0.0 in /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages (from ansible-core~=2.14.0) (3.1.3)
Requirement already satisfied: PyYAML>=5.1 in /home/guido/Development/virtualenv/2.14/lib64/python3.11/site-packages (from ansible-core~=2.14.0) (6.0.1)
Requirement already satisfied: cryptography in /home/guido/Development/virtualenv/2.14/lib64/python3.11/site-packages (from ansible-core~=2.14.0) (42.0.5)
Requirement already satisfied: packaging in /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages (from ansible-core~=2.14.0) (24.0)
Requirement already satisfied: resolvelib<0.9.0,>=0.5.3 in /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages (from ansible-core~=2.14.0) (0.8.1)
Requirement already satisfied: MarkupSafe>=2.0 in /home/guido/Development/virtualenv/2.14/lib64/python3.11/site-packages (from jinja2>=3.0.0->ansible-core~=2.14.0) (2.1.5)
Requirement already satisfied: cffi>=1.12 in /home/guido/Development/virtualenv/2.14/lib64/python3.11/site-packages (from cryptography->ansible-core~=2.14.0) (1.16.0)
Requirement already satisfied: pycparser in /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages (from cffi>=1.12->cryptography->ansible-core~=2.14.0) (2.21)
Downloading ansible_core-2.14.14-py3-none-any.whl (2.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.2/2.2 MB 1.6 MB/s eta 0:00:00
Installing collected packages: ansible-core
  Attempting uninstall: ansible-core
    Found existing installation: ansible-core 2.16.4
    Uninstalling ansible-core-2.16.4:
      Successfully uninstalled ansible-core-2.16.4
Successfully installed ansible-core-2.14.14

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: pip install --upgrade pip
(2.14) guido@fedora:~/Development/redhat/common$ ansible --version
ansible [core 2.14.14]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/guido/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages/ansible
  ansible collection location = /home/guido/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/guido/Development/virtualenv/2.14/bin/ansible
  python version = 3.11.8 (main, Feb 28 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)] (/home/guido/Development/virtualenv/2.14/bin/python)
  jinja version = 3.1.3
  libyaml = True
```

and finally the successful:

```
(2.14) guido@fedora:~/Development/redhat/common$ time ansible-galaxy collection install --no-cache fedora.linux_system_roles --force -vvv
ansible-galaxy [core 2.14.14]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/guido/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/guido/Development/virtualenv/2.14/lib/python3.11/site-packages/ansible
  ansible collection location = /home/guido/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/guido/Development/virtualenv/2.14/bin/ansible-galaxy
  python version = 3.11.8 (main, Feb 28 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)] (/home/guido/Development/virtualenv/2.14/bin/python)
  jinja version = 3.1.3
  libyaml = True
Using /etc/ansible/ansible.cfg as config file
Starting galaxy collection install process
Found installed collection middleware_automation.redhat_csp_download:1.2.2 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/redhat_csp_download'
Found installed collection middleware_automation.janus:1.0.0 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/janus'
Found installed collection middleware_automation.jws:1.2.5 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/jws'
Found installed collection middleware_automation.jbcs:1.0.4 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/jbcs'
Found installed collection middleware_automation.infinispan:1.3.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/infinispan'
Found installed collection middleware_automation.keycloak:2.1.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/keycloak'
Found installed collection middleware_automation.wildfly:1.5.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/wildfly'
Found installed collection middleware_automation.amq:2.0.1 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/amq'
Found installed collection middleware_automation.common:1.2.0 at '/home/guido/.ansible/collections/ansible_collections/middleware_automation/common'
Found installed collection ansible.netcommon:5.1.1 at '/home/guido/.ansible/collections/ansible_collections/ansible/netcommon'
Found installed collection ansible.utils:2.10.3 at '/home/guido/.ansible/collections/ansible_collections/ansible/utils'
Found installed collection ansible.posix:1.5.4 at '/home/guido/.ansible/collections/ansible_collections/ansible/posix'
Found installed collection community.docker:3.4.8 at '/home/guido/.ansible/collections/ansible_collections/community/docker'
Found installed collection community.fqcn_migration:1.0.2 at '/home/guido/.ansible/collections/ansible_collections/community/fqcn_migration'
Found installed collection community.aws:7.1.0 at '/home/guido/.ansible/collections/ansible_collections/community/aws'
Found installed collection community.general:8.4.0 at '/home/guido/.ansible/collections/ansible_collections/community/general'
Found installed collection guidograzioli.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/kubernetes.kubevirt'
Found installed collection redhat.eap:1.3.4 at '/home/guido/.ansible/collections/ansible_collections/redhat/eap'
Found installed collection redhat.jbcs:1.0.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/jbcs'
Found installed collection redhat.sso:2.0.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/sso'
Found installed collection redhat.data_grid:1.3.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/data_grid'
Found installed collection redhat.rhbk:2.1.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/rhbk'
Found installed collection redhat.jws:2.1.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/jws'
Found installed collection redhat.amq_broker:2.0.1 at '/home/guido/.ansible/collections/ansible_collections/redhat/amq_broker'
Found installed collection redhat.runtimes_common:1.2.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/runtimes_common'
Found installed collection redhat.rhel_system_roles:1.23.0 at '/home/guido/.ansible/collections/ansible_collections/redhat/rhel_system_roles'
Found installed collection amazon.aws:7.3.0 at '/home/guido/.ansible/collections/ansible_collections/amazon/aws'
Found installed collection azure.azcollection:1.16.0 at '/home/guido/.ansible/collections/ansible_collections/azure/azcollection'
Found installed collection containers.podman:1.12.0 at '/home/guido/.ansible/collections/ansible_collections/containers/podman'
Found installed collection kubernetes.core:2.4.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/core'
Found installed collection guidograzioli.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old2'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt.old3'
Found installed collection kubevirt.core:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubernetes/kubevirt'
Found installed collection kubernetes.kubevirt:0.1.0 at '/home/guido/.ansible/collections/ansible_collections/guidograzioli/kubevirt'
Found installed collection kubevirt.core:1.1.0 at '/home/guido/.ansible/collections/ansible_collections/kubevirt/core'
Found installed collection skupper.network:1.2.1 at '/home/guido/.ansible/collections/ansible_collections/skupper/network'
Found installed collection infra.rhbk_datagrid_aws:1.0.0 at '/home/guido/.ansible/collections/ansible_collections/infra/rhbk_datagrid_aws'
Found installed collection fedora.linux_system_roles:1.75.3 at '/home/guido/.ansible/collections/ansible_collections/fedora/linux_system_roles'
Process install dependency map
Opened /home/guido/.ansible/galaxy_token
Starting collection install process
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/fedora-linux_system_roles-1.75.3.tar.gz to /home/guido/.ansible/tmp/ansible-local-2524389i8lb1fw2/tmp7hfdyt61/fedora-linux_system_roles-1.75.3-elyoquhp
Collection 'fedora.linux_system_roles:1.75.3' obtained from server default https://galaxy.ansible.com/api/
Installing 'fedora.linux_system_roles:1.75.3' to '/home/guido/.ansible/collections/ansible_collections/fedora/linux_system_roles'
fedora.linux_system_roles:1.75.3 was installed successfully
'ansible.posix:1.5.4' is already installed, skipping.
'community.general:8.4.0' is already installed, skipping.
'containers.podman:1.12.0' is already installed, skipping.

real	0m8.505s
user	0m1.289s
sys	0m0.191s
```

**Comment by richm** [comment]
I'm using fedora 38
```
pip install 'ansible-core==2.14.0'
...
Successfully installed MarkupSafe-2.1.5 PyYAML-6.0.1 ansible-core-2.14.0 cffi-1.16.0 cryptography-42.0.5 jinja2-3.1.3 packaging-24.0 pycparser-2.21 resolvelib-0.8.1

ansible-galaxy collection install --no-cache fedora.linux_system_roles --force -vvvvv 2>&1 | tee output
...
Calling Galaxy at https://galaxy.ansible.com/api/v3/collections/fedora/linux_system_roles/versions/?limit=100
Calling Galaxy at https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/index/fedora/linux_system_roles/versions/?limit=100&offset=100
Calling Galaxy at https://galaxy.ansible.com/api/v3/collections/fedora/linux_system_roles/versions/1.9.2/
... huge long list of many, many system roles collection releases ...
... this takes a few minutes ...
Fetching a collection tarball for 'fedora.linux_system_roles:1.75.3' from Ansible Galaxy
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/fedora-linux_system_roles-1.75.3.tar.gz to /home/rmeggins/.ansible/tmp/ansible-local-6340620nhe1yj4/tmp0rrjzf4o/fedora-linux_system_roles-1.75.3-auh93xym
Validating downloaded file hash ff0e5cf72f58d7b7be7f0c4ca55f0932e7b78525b969422628db711dab0fc0db with expected hash ff0e5cf72f58d7b7be7f0c4ca55f0932e7b78525b969422628db711dab0fc0db
Collection 'fedora.linux_system_roles:1.75.3' obtained from server default https://galaxy.ansible.com/api/
Installing 'fedora.linux_system_roles:1.75.3' to '/home/rmeggins/.ansible/collections/ansible_collections/fedora/linux_system_roles'
fedora.linux_system_roles:1.75.3 was installed successfully
```
So I think the issues are
* there are too many fedora.linux_system_roles releases for poor Galaxy pulp to handle - we should see if we can remove some of those older releases
* ansible-galaxy 2.14.0 did a very inefficient search for the latest release - later versions improved this greatly

So I don't think 2.14.0 is "stuck" or "hung", I think it is just taking a very long time.  If I use `strace` on the python process, I can see it doing many, many Galaxy API http calls.
